cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -55,6 +56,32 @@ visualization_msgs::Marker CreateTrajectoryMarker(const int trajectory_id,
   return marker;
 }
 
+int GetLandmarkIndex(
+    const std::string& landmark_id,
+    std::unordered_map<std::string, int>* landmark_id_to_index) {
+  auto it = landmark_id_to_index->find(landmark_id);
```

Comment:
Why not just use

```c++
const int new_index = landmark_id_to_index->size()++;
landmark_id_to_index->emplace(std::make_pair(landmark_id, new_index));
return landmark_id_to_index->at(landmark_index);
```
Should get rid of the if condition.
http://en.cppreference.com/w/cpp/container/unordered_map/emplace

(not tested)

Reply:
I think it is less readable.